source/sidechains/creating-your-own.md

```diff
@@ -3,162 +3,112 @@ title: Building A New Sidechain with Elements
+    * ensure that the output of combineblocksigs has "complete" true
+  > `bc getblockcount`
+    * check the current block count
+  > `bc submitblock $SIGNED_BLOCK`
+  > `bc getblockcount`
+    * check that the chain advanced by one block
+
+Block generation can be easily automated in a shell script.
+
+#### Congratulations!
+Your chain is online and already accessible, just let other nodes connect to you:
+```
+bc addnode <your_url>
+```
+
+### Moving Bitcoin into your Sidechain
+You might have figured out that while you are in `regtest` mode you own all
+coins on the elements chain. Of course, in production you first have to use a
+mechanism known as peg-in to move coins from the main chain to the elements
+sidechain. Let's set up our own sidechain federation to handle the peg.
+
+1. Ensure that you have bitcoind 0.13.1 installed and elements as well as bitcoin are shut down.
+2. Elements fully validates that peg-in transactions exists in Bitcoin. Therefore, it opens a RPC connection to bitcoind and checks that a particular block is in the chain and has the required confirmations. We need to ensure that the elements daemon is correctly talking to Bitcoin.
+    * The configuration of Elements is getting larger, so we best create a config file in a new data directory.
+    * `BETADATADIR=/tmp/elementsdatadir/`
+    * `mkdir $BETADATADIR`
+    * Put the following into $BETADATADIR/elements.conf
+        ```
+        rpcuser=bitcoinrpc
+        rpcpassword=password
+        rpcport=8339
+        daemon=1
+        discover=0
+        testnet=0
+        regtest=1
+
+        mainchainrpchost=127.0.0.1
+        mainchainrpcport=8338
+        mainchainrpcuser=bitcoinrpc
+        mainchainrpcpassword=password
+
+        validatepegin=1
+        txindex=1
+        ```
+    * Now do the same for bitcoin
+    * `BITCOINDATADIR=/tmp/bitcoindatadir/`
+    * `mkdir $BITCOINDATADIR`
+    * put the following into $BITCOINDATADIR/bitcoin.conf
+        ```
+        rpcuser=bitcoinrpc
+        rpcpassword=password
+        rpcport=8338
+        daemon=1
+        discover=0
+        testnet=0
+        regtest=1
+        txindex=1
+        ```
+* In the previous section we've used our own public key for the blocksigner script. We will use the same for the federation script. This is the script that owns the coins on the sidechain.
+* Let's start bitcoin `bitcoind -datadir=$BITCOINDATADIR` and elements `elementsd -datadir=$BETADATADIR -signblockscript=$SIGNBLOCKSCRIPT -fedpegscript=$SIGNBLOCKSCRIPT`
+* Let's update the elements-cli alias to use the new datadir
+    * `alias bc="elements-cli -datadir=$BETADATADIR"`
+    * `alias bitc="bitcoin-cli -datadir=$BITCOINDATADIR"`
+* Create some coins on the main chain
+    * `bitc generate 101`
+* We can't create a regular peg-in yet, because there are no WPV outputs in regtest mode. We first create them by `sendtomainchain <addr>`. Note that peg-outs don't work, because you have no pegoutwatcher running.
```

Comment:
This was my fault, the API is slightly different from Liquid to here. It was looking for a "elements" style address instead of Bitcoin. They are different, in contrast to alpha. I pushed a fix for the PR if you want to try.